example-app/src/main/java/com/adyen/checkout/example/ui/configuration/CheckoutConfigurationProvider.kt

```diff
@@ -49,65 +46,53 @@ internal class CheckoutConfigurationProvider @Inject constructor(
-            .setInstallmentConfigurations(getInstallmentConfiguration())
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    private fun getCashAppPayConfiguration(): CashAppPayConfiguration =
-        CashAppPayConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setReturnUrl(CashAppPayComponent.getReturnUrl(context))
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    fun getBlikConfiguration(): BlikConfiguration =
-        BlikConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    fun getBacsConfiguration(): BacsDirectDebitConfiguration =
-        BacsDirectDebitConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    fun getInstantConfiguration(): InstantPaymentConfiguration =
-        InstantPaymentConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    fun getGiftCardConfiguration(): GiftCardConfiguration =
-        GiftCardConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .setPinRequired(true)
-            .build()
+    val checkoutConfig: CheckoutConfiguration
+        get() = CheckoutConfiguration(
+            shopperLocale = shopperLocale,
+            environment = environment,
+            clientKey = clientKey,
+            amount = amount,
+            analyticsConfiguration = getAnalyticsConfiguration(),
+        ) {
+            // Drop-in
+            dropIn {
+                setEnableRemovingStoredPaymentMethods(true)
+            }
+
+            // Payment methods
+            bcmc {
+                setShopperReference(keyValueStorage.getShopperReference())
+                setShowStorePaymentField(true)
+            }
+
+            card {
+                setShopperReference(keyValueStorage.getShopperReference())
+                setAddressConfiguration(getAddressConfiguration())
+                setInstallmentConfigurations(getInstallmentConfiguration())
+            }
+
+            cashAppPay {
+                setReturnUrl(CashAppPayComponent.getReturnUrl(context))
+            }
+
+            giftCard {
+                setPinRequired(true)
+            }
+
+            googlePay {
+                setCountryCode(keyValueStorage.getCountry())
+            }
+
+            // Actions
+            adyen3DS2 {
+                setThreeDSRequestorAppURL("https://www.adyen.com")
+            }
```

Comment:
Currently I am able to call methods of every available implicit receiver inside a lambda, like:
```
adyen3DS2 {
    adyen3DS2 {
        adyen3DS2 {
            ...
        }
    }
}
```

It can be limitted by using a `@DslMarker` annotation.

You can create a new annotation:
```
@DslMarker
@Target(AnnotationTarget.CLASS, AnnotationTarget.TYPE)
annotation class CheckoutConfigurationMarker
```

and apply the `@CheckoutConfigurationMarker` annotation to `CheckoutConfiguration` class and to all extension function receivers like:

```
fun CheckoutConfiguration.adyen3DS2(
    configuration: (@CheckoutConfigurationMarker Adyen3DS2Configuration.Builder).() -> Unit = {}
): CheckoutConfiguration {
    ...
}
```

After that this should be the result:
![Screenshot 2024-01-26 at 10 50 27](https://github.com/Adyen/adyen-android/assets/8832594/e041f98d-df8a-47e3-84ed-4a6900da04a8)

Reply:
That's really nice! I was wondering before if something like this was possible